flexget/components/notify/notifiers/mqtt.py

```diff
@@ -0,0 +1,204 @@
+            'broker_transport': {'type': 'string', 'default': 'tcp', 'enum': ['tcp', 'websockets']},
+            'broker_protocol': {'type': 'string', 'default': 'MQTTv311', 'enum': ['MQTTv31', 'MQTTv311']},
+            'username': {'type': 'string', 'default': ''},
+            'password': {'type': 'string', 'default': ''},
+            'enable_encrypted_communication': {'type': 'boolean', 'default': False},
+            'certificates': {
+                'type': 'object',
+                'properties': {
+                    'broker_ca_cert': {'type': 'string', 'default': ''},
+                    'client_cert': {'type': 'string', 'default': ''},
+                    'client_key': {'type': 'string', 'default': ''},
+                    'validate_broker_cert': {'type': 'boolean', 'default': True},
+                    'tls_version': {'type': 'string', 'default': '', 'enum': ['tlsv1.2', 'tlsv1.1', 'tlsv1', '']},
+                },
+                'additionalProperties': False,
+            },
+            'qos': {'type': 'integer', 'minimum': 0, 'maximum': 2, 'default': 0},
+            "retain": {"type": "boolean", 'default': False},
+        },
+        'additionalProperties': False,
+        'required': ['broker_address','topic'],
+    }
+
+    def notify(self, title, message, config):
+        """
+        Publish to an MQTT topic
+        """
+        try:
+            import paho.mqtt.client as mqtt
+            from hashlib import sha256
+            import ssl
+            import time
+
+        except ImportError as e:
+            logger.verbose('Error importing paho.mqtt.client: {}', e)
+            raise DependencyError(
+                plugin_name, 'paho.mqtt.client', 'paho-mqtt python module is required for MQTT notify plugin. ImportError: %s' % e
+            )
+
+        config['title'] = title
+        config['message'] = message
+        config['payload'] = '{} - {}'.format(config.get('title'),config.get('message'))
+
+        class PublishMQTT(mqtt.Client):
+            conn_rc_description_map = { 0: 	'Connection Accepted',
+                                1: 	'Connection Refused, unacceptable protocol version - The Server does not support the level of the MQTT protocol requested by the Client',
+                                2: 	'Connection Refused, identifier rejected - The Client identifier is correct UTF-8 but not allowed by the Server',
+                                3: 	'Connection Refused, Server unavailable - The Network Connection has been made but the MQTT service is unavailable',
+                                4: 	'Connection Refused, bad user name or password - The data in the user name or password is malformed',
+                                5: 	'Connection Refused, not authorized - The Client is not authorized to connect' }
+
+            MQTT_proto_map = { 'MQTTv31' : mqtt.MQTTv311,
+                               'MQTTv311': 	mqtt.MQTTv31 }
+
+            def __init__(self, config):
```

Comment:
creating a class and having its entire logic occur on init is a code smell that you probably do not even need a class...

Reply:
Good point.  I was using some example code to start, so I refactored to remove the excess class with commit 964dba7624322c86701376d4875a23025a9b8bf8